app/celery/celery.py

```diff
@@ -0,0 +1,30 @@
+from celery import Celery, Task
+
+
+class NotifyCelery(Celery):
+
+    def init_app(self, app):
+        # this task is nested so that it has access to the original flask application - when celery restarts processes,
+        # it doesn't appear to execute the whole import process again, and as such, flask.current_app no longer has
+        # a context, and throws a RuntimeError. So we need to create an app context from scratch each time.
+        class NotifyTask(Task):
+            abstract = True
+
+            def on_failure(self, exc, task_id, args, kwargs, einfo):
+                # ensure task will log exceptions to correct handlers
+                with app.app_context():
+                    app.logger.exception('Celery task failed')
```

Comment:
can you chuck `self.name` in to this log message to make debugging a lil' quicker?

eg `Celery task {self.name} failed`

Reply:
Good suggestion, I've made that change